docs/spec/currency.rst

```diff
@@ -0,0 +1,515 @@
+.. _currency:
+
+Currency
+========
+
+Overview
+~~~~~~~~
+
+This currency pallet provides an interface for the other pallets to manage balances of different currencies. It is a wrapper around the the `orml-tokens <https://github.com/open-web3-stack/open-runtime-module-library>`_ pallet. As such, accounts have two balances per currency: they have a ``reserved`` amount and a ``free`` amount. Users are able to freely transfer ``free`` balances, but only the parachain pallets are able to operate on ``reserved`` amounts.
+
+The external API for dispatchable and RPC functions use 'thin' amount types, meaning that the used currency depends on the context. For example, the currency used in :ref:`depositCollateral` depends on the vault's ``currencyId``. Sometimes, as is for example the case for :ref:`registerVault`, the function takes an additional ``currencyId`` argument to specify the currency to use. In contrast, internally in the parachain amounts are often represented by the ``Amount`` type defined in this pallet, which in addition to the amount, also contains the used currency. The benefit of this type is two-fold. First, we can guarantee that operations only work on compatible amounts. For example, it prevents adding DOT amounts to KSM amounts. Second, it allows for a more convenient api.
+
+Data Model
+~~~~~~~~~~
+
+Structs
+-------
+
+Amount
+......
+
+Stores an amount and the used currency.
+
+.. tabularcolumns:: |l|l|L|
+
+======================  ==========  =======================================================	
+Parameter               Type        Description                                            
+======================  ==========  =======================================================
+``amount``              Balance     The amount.
```

Comment:
I think we should name this `balance` to distinguish this from the outer type.

Reply:
Agree